### DISCUSSION TITLE
Adding express like middleware

### DISCUSSION THREAD

**Discussion by veritem** [vercel/next.js#18612] [Help] [answered]
I'm converting an existing server to use Next api routes and Typescript

I would like to ask if there is a way i can handle custom middlewares just like we do in express 

for thing like protecting api routes

**Comment by veritem** [comment]
Is there any one who have did this before

**Reply by HamzaAmar** [reply]
@makuzaverite You can work with this library it can help you 
[library](https://www.npmjs.com/package/next-connect)
[article](https://hoangvvo.com/blog/nextjs-middleware)

**Reply by veritem** [reply]
I'm going to git it a try

**Comment by leerob** [comment, marked as answer]
Yes, check out the documentation on [API Middlewares](https://nextjs.org/docs/api-routes/api-middlewares).
